gulpfile.js

```diff
@@ -397,7 +397,7 @@ function createSandboxExternal(defines) {
     transform("utf8", content => {
       content = preprocessor2.preprocessPDFJSCode(ctx, content);
       return `${licenseHeader}\n${content}`;
-    })
+    }).pipe(rename("pdf.sandbox.external.jsm"))
   );
```

Comment:
Did you forget to diff the output of `gulp mozcentral` against the master branch with this patch :-P
Because it doesn't actually work, and building breaks, since what you need is here is the following :-)
```suggestion
    })
  )
  .pipe(rename("pdf.sandbox.external.jsm"));
```

---

What happened here is that you placed the new `pipe`-call after the `transform`-call, which doesn't work, and you rather need to chain the `pipe`-calls themselves after one another.

Reply:
It works for me... or am I overlooking something:
```bash
➜  pdf.js.calixteman git:(mc) ✗ rm -fr build && gulp mozcentral && ls build/mozcentral/browser/extensions/pdfjs/content/build
[10:30:36] Using gulpfile ~/dev/mozilla/pdf.js.calixteman/gulpfile.js
[10:30:36] Starting 'mozcentral'...
[10:30:36] Starting 'createBuildNumber'...

### Getting extension build number
Extension build number: 74
[10:30:36] Finished 'createBuildNumber' after 22 ms
[10:30:36] Starting 'scriptingMozcentral'...

### Building default preferences
[10:30:36] Finished 'scriptingMozcentral' after 394 ms
[10:30:36] Starting 'createMozcentral'...

### Building mozilla-central extension
[10:30:42] asset pdf.js 354 KiB [emitted] [big] (name: main)

webpack 5.72.1 compiled successfully
[10:30:42] asset pdf.scripting.js 105 KiB [emitted] (name: main)

webpack 5.72.1 compiled successfully
[10:30:44] asset viewer.js 371 KiB [emitted] [big] (name: main)

webpack 5.72.1 compiled successfully
[10:30:47] asset pdf.worker.js 1.83 MiB [emitted] [big] (name: main)

webpack 5.72.1 compiled successfully
[10:30:47] Finished 'createMozcentral' after 10 s
[10:30:47] Finished 'mozcentral' after 11 s
pdf.js  pdf.sandbox.external.jsm  pdf.scripting.js  pdf.worker.js
```

Reply:
anyway I agree with you the `pipe` shouldn't be there: I'm fixing.

Reply:
> It works for me... or am I overlooking something:

Please look at the *actual file contents*, and compare with the `master` branch, and you'll see where things break.